jenkins/PerfCI_Chaos/04_PerfCI_Chaos_Upgrade_OpenShift_Deployment/Jenkinsfile

```diff
@@ -113,30 +118,28 @@ END
             }
         }
 
-        stage('Prepare Google credential files') {
+        stage('Prepare Google Credential Files') {
             steps {
-                    script {
-                        // Assign the file path to the environment variable
-                        env.GOOGLE_DRIVE_CREDENTIALS = "${env.WORKSPACE}/${CREDENTIALS_FILE}"
-                        env.GOOGLE_DRIVE_TOKEN = "${env.WORKSPACE}/${TOKEN_FILE}"
+                script {
+                    // Create a unique temp directory
+                    env.TEMP_DIR = sh(script: "umask 77; mktemp -d -p '${env.WORKSPACE_PATH}'", returnStdout: true).trim()
+                    if (!env.TEMP_DIR) {
+                        error "Failed to create temporary directory!"
+                    }
 
-                        sh """
-                        echo "Checking if ${env.GOOGLE_DRIVE_CREDENTIALS} exists..."
-                        if [[ -f "${env.GOOGLE_DRIVE_CREDENTIALS}" ]]; then
-                            echo "File exists. Deleting it..."
-                            rm -f "${env.GOOGLE_DRIVE_CREDENTIALS}"
-                        fi
-                        echo "Checking if ${env.GOOGLE_DRIVE_TOKEN} exists..."
-                        if [[ -f "${env.GOOGLE_DRIVE_TOKEN}" ]]; then
-                            echo "File exists. Deleting it..."
-                            rm -f "${env.GOOGLE_DRIVE_TOKEN}"
-                        fi
+                    // Set file paths
+                    env.GOOGLE_DRIVE_CREDENTIALS = "${env.TEMP_DIR}/${GOOGLE_CREDENTIALS_FILE}"
+                    env.GOOGLE_DRIVE_TOKEN = "${env.TEMP_DIR}/${GOOGLE_TOKEN_FILE}"
 
-                        echo "Copying secret file to the temporary local file..."
-                        cp "$GOOGLE_DRIVE_CREDENTIALS_FILE" "${env.GOOGLE_DRIVE_CREDENTIALS}"
-                        cp "$GOOGLE_DRIVE_TOKEN_FILE" "${env.GOOGLE_DRIVE_TOKEN}"
-                        """
-                    }
+                    // Prepare credential files
+                    sh """
+                        set -e
+                        umask 77
+                        cp '${GOOGLE_DRIVE_CREDENTIALS_FILE}' '${env.GOOGLE_DRIVE_CREDENTIALS}' || error "Failed to copy ${GOOGLE_DRIVE_CREDENTIALS_FILE}"
+                        cp '${GOOGLE_DRIVE_TOKEN_FILE}' '${env.GOOGLE_DRIVE_TOKEN}' || error "Failed to copy ${GOOGLE_DRIVE_TOKEN_FILE}"
+                        chmod 700 '${env.GOOGLE_DRIVE_CREDENTIALS}' '${env.GOOGLE_DRIVE_TOKEN}' || error "Failed to set permissions on credential files"
```

Comment:
Files should be mode 600 rather than 700 unless they're intended to be executed.  Directories do need to be 700.  A better way to do this would be

```
chmod go-rwx '${env.GOOGLE_DRIVE_CREDENTIALS}' '${env.GOOGLE_DRIVE_TOKEN}'
```